src/preflight/internal-links.js

```diff
@@ -14,12 +14,12 @@ import { JSDOM } from 'jsdom';
 
 /**
  * Preflight check for internal links
- * @param scrapedObjects - Array of objects containing the URL and scraped data
- * @param pageAuthToken - Authentication token for the page
-  * @param context - Context object containing the logger
+ * @param {Array<Object>} scrapedObjects - Array of objects containing the URL and scraped data
+ * @param {Object} context - Context object containing the logger
+ * @param {RequestOptions} options - Options for to pass to the fetch request
  * @returns {Promise<Array>} - Array of objects containing the page URL and internal link status
  */
-export async function runInternalLinkChecks(scrapedObjects, pageAuthToken, context) {
+export async function runInternalLinkChecks(scrapedObjects, context, options = {}) {
```

Comment:
```suggestion
export async function runInternalLinkChecks(scrapedObjects, context, options = {
  pageAuthToken: null,
}) {
```
I would prefer to be more specific here: state that `options` can optionally include `pageAuthToken` and set the Authorization header if the token is provided.